packages/jss/src/index.d.ts

```diff
@@ -7,35 +7,38 @@ import {Properties as CSSProperties} from 'csstype'
 // TODO: refactor to only include Observable types if plugin is installed.
 import {Observable} from 'indefinite-observable'
 
-// TODO: Type data better, currently typed as any for allowing to override it
-type Func<R> = ((data: any) => R)
+type Func<P, T, R> = T extends undefined ? ((data: P) => R) : ((data: P & {theme: T}) => R)
```

Comment:
maybe ThemeFunc is a better name then.

Reply:
Yeah the name here is a hard one. The `theme` type isn't always guaranteed to be provided, so I don't know if `ThemeFunc` would work. If a prefix would add clarity, maybe something like `DataFunc` or `StylesFunc`? Just some ideas. Other options could exist too.

Reply:
Just to make it easier to keep track of everything, I'm gonna resolve this for now. (If that's dissatisfactory, feel free to unresolve this conversation.)

Renaming `Func` isn't necessarily related to this PR and would create some noise for anyone reviewing the changes in the future. So I'll leave the naming improvement to the future.